doc/howto/write_a_view/solution_iterator.cpp

```diff
@@ -0,0 +1,95 @@
+//![start]
+#include <iostream>
+#include <vector>
+#include <seqan3/alphabet/nucleotide/all.hpp>
+#include <seqan3/std/ranges>
+
+using namespace seqan3;
+
+template <std::ranges::ForwardRange urng_t> // the underlying range type
+struct my_iterator : std::ranges::iterator_t<urng_t>
+{
+//![start]
+    //![static_assert]
+    static_assert(NucleotideAlphabet<reference_t<urng_t>>,
```

Comment:
```suggestion
    static_assert(NucleotideAlphabet<delete_const_t<reference_t<urng_t>>>,
```

If you want it to work on const containers
Also adapt in other snipputets if you want to

Reply:
will be resolved by alphabet refactor